### PR TITLE
Implement golangci-lint for pipeline lint test 

### PR DESCRIPTION
**Pull request by jakeandandrews**
## What
Golangci-lint was implemented to test for correct linting within the pipeline. The code was altered to keep it in line with the new linting rules. Additionally component tests were handled in thr MakeFile similarly to how they are set up here : https://github.com/ONSdigital/dp-frontend-cookie-controller/blob/develop/Makefile

## How to review
When future PRs are created, the pipeline will test for correct lint

## Who can review
A member of ONS

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        - revive
+      source: "github.com/smartystreets/goconvey/convey"
+  new: false
+
+# golangci.com configuration
+# https://github.com/golangci/golangci/wiki/Configuration
+service:
+  golangci-lint-version: 1.55.x # use the fixed version to not introduce new linters unexpectedly
+  prepare:
+    - echo "here I can run custom commands, but no preparation needed for this repo"
```
Comment (jmgq) [inline .golangci.yml]: Small one, but all the new files in this PR are missing a new line at the end of the file. You can configure your IDE to add it automatically.

File: Makefile [hunk 2]
```diff
@@ -33,3 +33,16 @@ debug:
...
+	golangci-lint run ./...
+
+.PHONY: lint-local
+lint-local: ## Use locally to run linters against Go code
+	go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.55.2
+	golangci-lint run ./...
+
+.PHONY: test-component
+test-component:
+	exit 0
```
Comment (lindenmckenzie) [inline Makefile]: Could add a comment here to say why an exit command has been given.

File: api/api.go [hunk 3]
```diff
@@ -146,6 +136,7 @@ func handleDownload(handler func(r *http.Request) (io.ReadCloser, map[string]str
 			if err != nil {
 				log.Error(ctx, "handleDownload: Error while copying from reader", err, log.Data{"request:": request})
 				http.Error(w, err.Error(), http.StatusInternalServerError)
+				return
```
Comment (lindenmckenzie) [inline api/api.go:139]: Not sure about this - there's only a return statement in this branch. Presuming the linter complained?

File: cmd/dp-file-downloader/main.go [hunk 4]
```diff
@@ -8,7 +8,7 @@ import (
 	"os/signal"
 	"syscall"
 
-	healthcheck "github.com/ONSdigital/dp-api-clients-go/health"
+	healthcheckapi "github.com/ONSdigital/dp-api-clients-go/health"
```
Comment (lindenmckenzie) [inline cmd/dp-file-downloader/main.go]: I'm not sure about this name change - it's not an api, but a healthcheck for other apis. What complaint did the linter have about it being called healthcheck?

File: table/clients.go [hunk 5]
```diff
@@ -6,12 +6,12 @@ import (
 )
 
 //go:generate moq -out testdata/zebedeeclient.go -pkg testdata . ZebedeeClient
-//go:generate moq -out testdata/tablerendererclient.go -pkg testdata . TableRendererClient
+//go:generate moq -out testdata/tablerendererclient.go -pkg testdata . RendererClient
```
Comment (lindenmckenzie) [inline table/clients.go:10]: Why this name change?

File: table/table_test.go [hunk 6]
```diff
@@ -9,16 +9,22 @@ import (
...
-const renderHost = "render"
+var requestURI = "/foo/bar.json"
+var requestFormat = "html"
+var expectedDisposition = "attachment; filename=\"bar.html\""
+var accessToken = "myAccessToken"
+var uriParam = "&uri="
+var expectedContentType = "text/html"
+var expectedContent = "renderServerResponse"
+var contentServerResponse = "contentServerResponse"
+var baseURL = "http://localhost/download/table?format="
```
Comment (lindenmckenzie) [inline table/table_test.go]: Could be a var block or use const for these but not massively fussed.

**Review by lindenmckenzie — COMMENTED** [review]
Few minor comments, happy to discuss as may be happy to pass as is.

File: api/api.go (hunk 3 above)
Comment (jakeandandrews) [reply api/api.go:139]: the linter complained, yep

File: cmd/dp-file-downloader/main.go (hunk 4 above)
Comment (jakeandandrews) [reply cmd/dp-file-downloader/main.go]: The linter complained that healthcheck was defined here on line 11 as an import and then also on line 61 as a var declaration. Had to change one of the names.

File: table/clients.go (hunk 5 above)
Comment (jakeandandrews) [reply table/clients.go:10]: Again, the linter complained - this time that having TableRendererClient might lead to table.TableRendererClient (which it didn't like) and recommended renaming to RendererClient so it be table.RendererClient.

File: Makefile (hunk 2 above)
Comment (jakeandandrews) [reply Makefile]: sure, I will add that this is in lieu of fleshed out component tests

File: table/table_test.go (hunk 6 above)
Comment (jakeandandrews) [reply table/table_test.go]: I may as well do a var block - keeps it all nice and tidy

File: cmd/dp-file-downloader/main.go (hunk 4 above)
Comment (lindenmckenzie) [reply cmd/dp-file-downloader/main.go]: Maybe `healthcheckclient` then

**Review by lindenmckenzie — APPROVED** [review]

**Pull request merged**